psci/PSCi/Types.hs

```diff
@@ -121,12 +121,17 @@ updateModules modules st =
 
 -- |
 -- Updates the state to have more let bindings.
+-- Replaces ValueDeclarations if they have the same identifier.
 --
 updateLets :: [P.Declaration] -> PSCiState -> PSCiState
-updateLets ds st = st { psciLetBindings = psciLetBindings st ++ ds }
-
+updateLets ds st = st { psciLetBindings = keptBindings ++ ds }
+  where keptBindings = filter shouldKeep (psciLetBindings st)
+        shouldKeep decl = valueDeclarationIdent decl `notElem` map valueDeclarationIdent ds
```

Comment:
Not sure what to key against. If it were just value declarations I'd use ```Ident```, but ```psciLetBindings```can include data declarations or types as well (it looks misnamed to me).